site/gatsby-site/src/components/taxa/TaxonomyForm.js

```diff
@@ -60,275 +29,208 @@ const TaxonomyForm = forwardRef(function TaxonomyForm({ namespace, incidentId, o
 
   const [fieldsWithDefaultValues, setFieldsWithDefaultValues] = useState([]);
 
-  const [taxonomy, setTaxonomy] = useState(null);
-
-  const { runQuery } = useMongo();
+  const [deletedSubClassificationIds, setDeletedSubClassificationIds] = useState([]);
 
   const addToast = useToastContext();
 
   const formRef = useRef(null);
 
+  const debouncedSetInitialValues = useRef(
```

Comment:
I found that when two taxonomies were open for editing at the same time, changing responses for one would sometimes change them for the other as well. To eliminate that class of errors, I set it so that only one taxonomy form could be rendered at a time. However, that means that that if you enter some values for taxonomy A, then hit edit to look at taxonomy B, when you return to taxonomy A, the values you entered are no longer there. To fix that, I needed to make it so the initial values are always set to whatever the most recently entered values were. That has to be debounced, though, because otherwise it makes typing laggy.

Reply:
That's really weird! Looks like this might cause a race condition if whatever you're waiting for takes longer than `500`. You can try with `0`, which is a bit less race-condition-prone. 

But eventually, we should fix it properly, which I think will require moving some `fetch` and `useEffect` around.